sdk/core/Azure.Core/src/OperationFailedException.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Azure.Core;
+
+namespace Azure
+{
+    /// <summary>
+    /// An exception thrown when service request fails.
+    /// </summary>
+    [Serializable]
+    public class OperationFailedException : RequestFailedException
+    {
+        /// <summary>
+        /// Gets the Id of the related <see cref="Operation{T}"/>/>
+        /// </summary>
+        public string? OperationId { get; }
```

Comment:
In which cases will the `OperationId` be null?

Reply:
There may not be a valid case.